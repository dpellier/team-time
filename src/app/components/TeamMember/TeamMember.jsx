
import React from 'react';
import PropTypes from 'prop-types';
import defaultPicture from '../../../assets/default-member.png';

const TeamMember = ({ name, picture }) =>
    <div className='team-member'>
        <div className='team-member__profile'>
            <img className='team-member__profile__picture'
                 src={ picture }
                 onError={ (e) => { e.target.src = defaultPicture; }} />

            <div className='team-member__profile__menu'>
                {/* Real links not implemented */}
                <span className='team-member__profile__menu__item'></span>
                <span className='team-member__profile__menu__item'></span>
                <span className='team-member__profile__menu__item'></span>
                <span className='team-member__profile__menu__item'></span>
            </div>
        </div>

        <span className='team-member__name'>
            { name }
        </span>
    </div>
;

TeamMember.propTypes = {
    name: PropTypes.string,
    picture: PropTypes.string
};

export { TeamMember };
