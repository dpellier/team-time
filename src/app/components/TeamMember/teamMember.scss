
@import '../../style/colors';

$team-member-width: 50px;
$team-member-height: 50px;
$team-member-menu-border-size: 1px;

.team-member {
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    text-align: center;

    &__profile {
        position: relative;
        width: $team-member-width;
        height: $team-member-height;

        &__picture {
            position: absolute;
            top: 1px;
            left: 1px;
            z-index: 2;
            border-radius: 50%;
            background-color: $white;
            width: $team-member-width - ($team-member-menu-border-size * 2);
            height: $team-member-height - ($team-member-menu-border-size * 2);
            box-sizing: border-box;
            pointer-events: none;
        }

        &__menu {
            position: absolute;
            top: 0;
            left: 0;
            transition: transform ease .3s;
            border: solid $team-member-menu-border-size $black;
            border-radius: 50%;
            background-color: $white;
            width: $team-member-width;
            height: $team-member-height;
            overflow: hidden;
            box-sizing: border-box;

            &:hover {
                transform: scale(1.8);
            }

            &__item {
                position: absolute;
                top: 0;
                left: 50%;
                background: $white no-repeat 3px 4px;
                background-size: 35%;
                cursor: pointer;
                width: $team-member-width / 2;
                height: $team-member-height / 2;
                clip-path: polygon(0 0, 0 100%, 100% 0);

                &:hover {
                    background-color: $dodger-blue;
                }

                &:nth-child(1) {
                    background-image: url('assets/slack.png');
                }

                &:nth-child(2) {
                    transform: rotate(45deg);
                    background-image: url('assets/go-to-meeting.png');
                    transform-origin: 0 100%;
                }

                &:nth-child(3) {
                    transform: rotate(90deg);
                    background-image: url('assets/skype.png');
                    transform-origin: 0 100%;
                }

                &:nth-child(4) {
                    transform: rotate(135deg);
                    background-image: url('assets/github.png');
                    transform-origin: 0 100%;
                }
            }
        }
    }

    &__name {
        font-size: 12px;
    }
}
