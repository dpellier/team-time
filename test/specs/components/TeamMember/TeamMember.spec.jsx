
import React from 'react';
import renderer from 'react-test-renderer';
import { TeamMember } from '../../../../src/app/components/TeamMember/TeamMember';

describe('<TeamMember />', () => {
    it('is correctly rendered', () => {
        const component = renderer.create(
            <TeamMember name={ 'Name' }
                        picture={ 'URL' } />
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});
