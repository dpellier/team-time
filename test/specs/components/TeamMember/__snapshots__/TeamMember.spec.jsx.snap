// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamMember /> is correctly rendered 1`] = `
<div
  className="team-member"
>
  <div
    className="team-member__profile"
  >
    <img
      className="team-member__profile__picture"
      onError={[Function]}
      src="URL"
    />
    <div
      className="team-member__profile__menu"
    >
      <span
        className="team-member__profile__menu__item"
      />
      <span
        className="team-member__profile__menu__item"
      />
      <span
        className="team-member__profile__menu__item"
      />
      <span
        className="team-member__profile__menu__item"
      />
    </div>
  </div>
  <span
    className="team-member__name"
  >
    Name
  </span>
</div>
`;
